garden_ai/app/create.py

```diff
@@ -0,0 +1,172 @@
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import List, Optional
+
+import rich
+import typer
+from garden_ai.client import GardenClient
+from rich.prompt import Prompt
+
+logger = logging.getLogger()
+
+LOCAL_STORAGE = Path("~/.garden/db/").expanduser()
+(LOCAL_STORAGE / "gardens").mkdir(parents=True, exist_ok=True)
+(LOCAL_STORAGE / "pipelines").mkdir(parents=True, exist_ok=True)
+
+
+def setup_directory(directory: Optional[Path]) -> Optional[Path]:
+    """
+    Validate the directory provided by the user, scaffolding with "pipelines/" and
+    "models/" subdirectories if possible (i.e. directory does not yet exist or
+    exists but is empty).
+    """
+    if directory is None:
+        return None
+
+    if directory.exists():
+        if list(directory.iterdir()):
+            logger.fatal("Directory must be empty if it already exists.")
+            raise typer.Exit(code=1)
```

Comment:
rather than code=1, maybe something more descriptive would be better

Reply:
having a command return 1 is conventional to indicate failure to the parent process; iirc this comes from the C macros `EXIT_SUCCESS` and `EXIT_FAILURE` being usually defined as 0 and 1 on unix-likes, though I think technically anything nonzero is supposed to be interpreted as failure.